src/yoroi-wallets/storage.ts

```diff
@@ -44,3 +44,124 @@ export const makeStorageWithPrefix = (prefix: string): Storage => {
+  if (!isValidPath(path)) throw new Error('invalid path')
+  const isRootStorage = storage.path === '/'
+  const isRootPath = path === '/'
+
+  const storagePath =
+    isRootStorage && isRootPath
+      ? '/'
+      : !isRootStorage && isRootPath
+      ? storage.path
+      : isRootStorage && !isRootPath
+      ? `${storage.path}${path}`
+      : `${storage.path}/${path}`
+
+  const prefixedStorage: PrefixedStorage = {
+    path: storagePath,
+    toAbsolutePath: (filename: string) => toAbsolutePath(storage.path, path, filename),
+
+    async getItem(filename: string) {
+      if (!isValidFilename(filename)) throw new Error('invalid filename')
+      const absolutePath = toAbsolutePath(storage.path, path, filename)
+      return AsyncStorage.getItem(absolutePath)
+    },
+    async multiGet(filenames: string[]) {
+      const absolutePaths = filenames.map((filename) => {
+        if (!isValidFilename(filename)) throw new Error('invalid filename')
+        return toAbsolutePath(storage.path, path, filename)
+      })
+      const items = await AsyncStorage.multiGet(absolutePaths)
+      return items.map(([absolutePath, file]) => [toFilename(absolutePath), file])
+    },
+
+    async setItem(filename: string, file: string) {
+      if (!isValidFilename(filename)) throw new Error('invalid filename')
+      const absolutePath = toAbsolutePath(storage.path, path, filename)
+      return AsyncStorage.setItem(absolutePath, file)
+    },
+    async multiSet(items: string[][]) {
+      const absolutePathItems = items.map(([filename, file]) => {
+        if (!isValidFilename(filename)) throw new Error('invalid filename')
+        return [toAbsolutePath(storage.path, path, filename), file]
+      })
+      return AsyncStorage.multiSet(absolutePathItems)
+    },
+
+    async removeItem(filename: string) {
+      if (!isValidFilename(filename)) throw new Error('invalid filename')
+      const absolutePath = toAbsolutePath(storage.path, path, filename)
+      return AsyncStorage.removeItem(absolutePath)
+    },
+    async multiRemove(filenames: string[]) {
+      const absolutePaths = filenames.map((filename) => {
+        if (!isValidFilename(filename)) throw new Error('invalid filename')
+        return toAbsolutePath(storage.path, path, filename)
+      })
+      return AsyncStorage.multiRemove(absolutePaths)
+    },
+    async clear() {
+      const allKeys = await this.getAllKeys()
+      return this.multiRemove(allKeys)
+    },
+
+    async getAllKeys() {
+      const allKeys = await AsyncStorage.getAllKeys()
+      const regex = isRootStorage && isRootPath ? new RegExp(`^/[^/]*$`) : new RegExp(`^${storagePath}/[^/]*$`)
+
+      return allKeys.filter((absolutePath) => absolutePath.match(regex)).map(toFilename)
+    },
+  }
+
+  return prefixedStorage
+}
+
+export const isValidPath = (path: string) => {
+  if (path.match(/[\\^$*+?.()|[\]{}]/g)) return false // contains escaping characters
```

Comment:
imho, chaging from deny to approve will make it easier to read I.e `!x.match(^[a-zA-Z0-9_\-]+$)`